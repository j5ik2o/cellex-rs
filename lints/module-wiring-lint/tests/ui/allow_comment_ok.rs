#![feature(register_tool)]
#![register_tool(module_wiring)]
#![warn(module_wiring::no_parent_reexport)]

pub mod child {
  pub mod sub {
    pub struct Thing;
    pub struct Other;
  }
}

// allow module_wiring::no_parent_reexport
pub use child::sub::Thing;

pub use child::sub::Other; // allow module_wiring::no_parent_reexport

fn main() {}
