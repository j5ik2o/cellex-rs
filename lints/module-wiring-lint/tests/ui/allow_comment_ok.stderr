warning: 末端モジュールの宣言では `mod` のみを使用してください
  --> $DIR/allow_comment_ok.rs:6:3
   |
LL | /   pub mod sub {
LL | |     pub struct Thing;
LL | |     pub struct Other;
LL | |   }
   | |___^
   |
   = note: ルール: 末端モジュールを公開したい場合は親で `mod` のみ宣言し、`pub use` で公開します
   = note: 詳細: 宣言 `pub mod sub {
               pub struct Thing;
               pub struct Other;
             }` に公開可視性が設定されています
   = help: 宣言から可視性修飾子を削除し、公開が必要なシンボルは末端モジュール内で `pub` を付けてください
   = note: `#[warn(no_parent_reexport)]` on by default

warning: 1 warning emitted

